opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -606,3 +607,108 @@ def delete_task(self, task_id: str) -> object:
             method="DELETE",
             url=API_URL,
         )
+
+    def _get_profile(self, payload: Optional[dict] = None):
+        """
+        Get the profile using the given payload.
+
+        :param payload: The payload to be used for getting the profile. Defaults to None.
+        :type payload: Optional[dict]
+        :return: The response from the server after performing the request.
+        :rtype: Any
+        """
+        validate_profile_input(None, payload)
+        return self._client.transport.perform_request(
+            method="GET", url=f"{ML_BASE_URI}/profile", body=payload
+        )
+
+    def _get_models_profile(
+        self, model_id: Optional[str] = "", payload: Optional[dict] = None
+    ):
+        """
+        Get the profile of a model.
+
+        Args:
+            model_id (str, optional): The ID of the model. Defaults to "".
+            payload (dict, optional): Additional payload for the request. Defaults to None.
+
+        Returns:
+            dict: The response from the API.
+        """
+        validate_profile_input(model_id, payload)
+
+        url = f"{ML_BASE_URI}/profile/models/{model_id if model_id else ''}"
+        return self._client.transport.perform_request(
+            method="GET", url=url, body=payload
+        )
+
+    def _get_tasks_profile(
+        self, task_id: Optional[str] = "", payload: Optional[dict] = None
+    ):
+        """
+        Retrieves the profile of a task from the API.
+
+        Parameters:
+            task_id (str, optional): The ID of the task to retrieve the profile for. Defaults to an empty string.
+            payload (dict, optional): Additional payload for the request. Defaults to None.
+
+        Returns:
+            dict: The profile of the task.
+
+        Raises:
+            ValueError: If the input validation fails.
+
+        """
+        validate_profile_input(task_id, payload)
+
+        url = f"{ML_BASE_URI}/profile/tasks/{task_id if task_id else ''}"
+        return self._client.transport.perform_request(
+            method="GET", url=url, body=payload
+        )
+
+    def get_profile(
+        self,
+        profile_type: str = "all",
+        ids: Optional[Union[str, List[str]]] = None,
+        request_body: Optional[dict] = None,
+    ) -> dict:
+        """
+        Get profile information based on the profile type.
+
+        Args:
+            profile_type: The type of profile to retrieve. Valid values are 'all', 'model', or 'task'. Default is 'all'.
```

Comment:
describe a bit more these options? like what happens if somebody selects "model"

Reply:
Added more details to the profile_type.